ddtrace/internal/datadog/profiling/dd_wrapper/src/code_provenance.cpp

```diff
@@ -14,9 +14,8 @@ namespace Datadog {
 void
 Datadog::CodeProvenance::postfork_child()
 {
-    get_instance().mtx.~mutex();            // Destroy the mutex
```

Comment:
Does destroying the mutex result in an UB?

Reply:
I think whether or not it's UB is UB.

When a thread "expires" because of a fork, it is _not_ destroyed.  So the part of the standard that discusses the fact that destroying a mutex held by a destroyed thread is UB doesn't apply.

idk lol